src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb

```diff
@@ -569,7 +569,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                 End If
 
                                 Dim stepOperand As IOperation = If(stepValue.IsConstant, DirectCast(stepValue, IOperation), New Temporary(SyntheticLocalKind.ForLoopStepValue, BoundFor, stepValue))
-                                statements.Add(New CompoundAssignment(controlReference, stepOperand, Semantics.Expression.DeriveAdditionKind(controlType), Nothing, stepValue.Syntax))
+                                statements.Add(New ExpressionStatement(controlReference, stepOperand, Semantics.Expression.DeriveAdditionKind(controlType), Nothing, stepValue.Syntax))
```

Comment:
@AlekseyTs @genlu once we get pass the part that CompoundAssigment and Assignment was just duplicated code which is actually ExpressionStatement (expression in the ExpressionStatement is either CompoundAssignement or Assignment, so looks like duplicated code are created just for the expression? I dont know. see this - https://github.com/dotnet/roslyn/pull/18318/files#diff-5e5ff960b3a6f83d52b5b20c6e12fcf2L47)

why the code do "statements.Add(/*New*/ CompoundAssignment" this and create new ExpressionStatement rather than getting it from somewhere (I dont know where I would get one), I dont know. I wasn't trying to change that logic.